src/material/slider/slider.ts

```diff
@@ -874,6 +874,12 @@ export class MatSlider
 
   /** Calculates the percentage of the slider that a value is. */
   private _calculatePercentage(value: number | null) {
+    // when min,max,value are all zero, the fraction becomes NAN, because we divide 0/0
+    // so we have to explicitely return 1 in this case
+    if (value === 0 && this.min === 0 && this.max === 0) {
```

Comment:
I think we should avoid not only expression 0/0 but division by 0 in general, because If both min and max values are (for instance) 2 then it will path the "if" guard but still it will lead to division by 0 and return Infinity which can impact further calculations as well. Also, if we assume that value, min, and max are equal (for instance) 2 it will also path the guard but eventually, it will end up with 0 / 0 (NaN)

Reply:
Yes definitely that makes sense, this guard won't save us from every case.
 I'll check between, numerator and dominator against 0 and not on min,max,value